netty

lesline opened the pull request.

He-Pin committed:
= tweak with alignments

He-Pin committed:
+ 添加第1章的注释

He-Pin committed:
+ 添加第2章的注释

He-Pin committed:
+ 添加第4章的注释

He-Pin committed:
+ 添加第5章的注释

He-Pin committed:
+ 第6章的中文注释

He-Pin committed:
+ 第7章的中文注释

He-Pin committed:
+ 第8章的中文注释

He-Pin committed:
+ 第9章的中文注释

He-Pin committed:
+ 第10章的中文注释

He-Pin committed:
+ 第11章的中文注释

He-Pin committed:
+ 第12章的中文注释

He-Pin committed:
+ 第13章的中文注释

He-Pin committed:
添加本书的内容说明

He-Pin committed:
+ 添加封面图

He-Pin committed:
! 更新编译信息

He-Pin committed:
= fix url

He-Pin committed:
= 修复对分支的描述

He-Pin committed:
Update README.md

添加链接

He-Pin committed:
Update README.md

fixup

He-Pin committed:
Update README.md

He-Pin committed:
! pom upgrade to 4.1.10

He-Pin committed:
remove 异步社区

He-Pin committed:
添加勘误

He-Pin committed:
更改文案

He-Pin committed:
更改文案

He-Pin committed:
添加翻译心得文档

He-Pin committed:
! pom update to 4.1.11

He-Pin committed:
add 签售信息

He-Pin committed:
remove 签售信息

He-Pin committed:
change to html

He-Pin committed:
update to 4.1.12

Commit:
update to 4.1.14

Commit:
update to 4.1.15

He-Pin committed:
! pom upgrade to netty 4.1.16

He-Pin committed:
! pom upgrade to netty 4.1.17

Commit:
Merge remote-tracking branch 'origin/ChineseVersion' into ChineseVersion

# Conflicts:
#	pom.xml

Commit:
! pom update to 4.1.18

Commit:
修改了jquery源来自国内的，不用翻墙

He-Pin committed:
Merge pull request #7 from JaryZhen/ChineseVersion

修改了jquery源来自国内的，不用翻墙

He-Pin committed:
! pom upgrade to netty 4.1.19

He-Pin commented:
@hainanzhong2010 @lesline  Please open issue in the forked project,I will sync back some updates to upstream here .

He-Pin commented:
@lesline The Chinese version have Chinese comment ,so it can't be merged in upstream ,sorry.

He-Pin committed:
! pom upgrade to netty 4.1.20

He-Pin committed:
! pom upgrade to netty 4.1.21

He-Pin committed:
Update pom.xml

He-Pin committed:
! pom upgrade to netty 4.1.23

MottoX committed:
fix comment

He-Pin committed:
Merge pull request #9 from MottoX/ChineseVersion

修正一处注释

He-Pin committed:
! pom upgrade to netty 4.1.25

He-Pin committed:
Update README.md

Commit:
! pom upgrade to netty 4.1.27

Commit:
! pom upgrade to netty 4.1.29

Commit:
! pom upgrade to netty 4.1.32

dependabot-support committed:
Bump maven-failsafe-plugin from 2.18.1 to 2.22.1

Bumps [maven-failsafe-plugin](https://github.com/apache/maven-surefire) from 2.18.1 to 2.22.1.
- [Release notes](https://github.com/apache/maven-surefire/releases)
- [Commits](https://github.com/apache/maven-surefire/compare/surefire-2.18.1...surefire-2.22.1)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
Bump maven-clean-plugin from 2.6.1 to 3.1.0

Bumps [maven-clean-plugin](https://github.com/apache/maven-clean-plugin) from 2.6.1 to 3.1.0.
- [Release notes](https://github.com/apache/maven-clean-plugin/releases)
- [Commits](https://github.com/apache/maven-clean-plugin/compare/maven-clean-plugin-2.6.1...maven-clean-plugin-3.1.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
Bump maven-surefire-plugin from 2.18.1 to 2.22.1

Bumps [maven-surefire-plugin](https://github.com/apache/maven-surefire) from 2.18.1 to 2.22.1.
- [Release notes](https://github.com/apache/maven-surefire/releases)
- [Commits](https://github.com/apache/maven-surefire/compare/surefire-2.18.1...surefire-2.22.1)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
[Security] Bump protobuf-java from 2.5.0 to 3.6.1

Bumps [protobuf-java](https://github.com/google/protobuf) from 2.5.0 to 3.6.1. **This update includes security fixes.**
- [Release notes](https://github.com/google/protobuf/releases)
- [Changelog](https://github.com/protocolbuffers/protobuf/blob/master/generate_changelog.py)
- [Commits](https://github.com/google/protobuf/compare/v2.5.0...v3.6.1)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
Bump junit from 4.11 to 4.12

Bumps [junit](https://github.com/junit-team/junit) from 4.11 to 4.12.
- [Release notes](https://github.com/junit-team/junit/releases)
- [Changelog](https://github.com/junit-team/junit4/blob/master/doc/ReleaseNotes4.12.md)
- [Commits](https://github.com/junit-team/junit/compare/r4.11...r4.12)

Signed-off-by: dependabot[bot] <support@dependabot.com>

He-Pin committed:
Merge pull request #10 from ReactivePlatform/dependabot/maven/org.apache.maven.plugins-maven-failsafe-plugin-2.22.1

Bump maven-failsafe-plugin from 2.18.1 to 2.22.1

He-Pin committed:
Merge pull request #11 from ReactivePlatform/dependabot/maven/org.apache.maven.plugins-maven-surefire-plugin-2.22.1

Bump maven-surefire-plugin from 2.18.1 to 2.22.1

He-Pin committed:
Merge pull request #12 from ReactivePlatform/dependabot/maven/org.apache.maven.plugins-maven-clean-plugin-3.1.0

Bump maven-clean-plugin from 2.6.1 to 3.1.0

He-Pin committed:
Merge pull request #13 from ReactivePlatform/dependabot/maven/com.google.protobuf-protobuf-java-3.6.1

[Security] Bump protobuf-java from 2.5.0 to 3.6.1

He-Pin committed:
Merge pull request #14 from ReactivePlatform/dependabot/maven/junit-junit-4.12

Bump junit from 4.11 to 4.12

Commit:
add travis

dependabot-support committed:
Bump maven-compiler-plugin from 3.3 to 3.8.0

Bumps [maven-compiler-plugin](https://github.com/apache/maven-compiler-plugin) from 3.3 to 3.8.0.
- [Release notes](https://github.com/apache/maven-compiler-plugin/releases)
- [Commits](https://github.com/apache/maven-compiler-plugin/compare/maven-compiler-plugin-3.3...maven-compiler-plugin-3.8.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
Bump exec-maven-plugin from 1.2.1 to 1.6.0

Bumps [exec-maven-plugin](https://github.com/mojohaus/exec-maven-plugin) from 1.2.1 to 1.6.0.
- [Release notes](https://github.com/mojohaus/exec-maven-plugin/releases)
- [Commits](https://github.com/mojohaus/exec-maven-plugin/compare/exec-maven-plugin-1.2.1...exec-maven-plugin-1.6.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
Bump maven-jar-plugin from 2.6 to 3.1.1

Bumps [maven-jar-plugin](https://github.com/apache/maven-jar-plugin) from 2.6 to 3.1.1.
- [Release notes](https://github.com/apache/maven-jar-plugin/releases)
- [Commits](https://github.com/apache/maven-jar-plugin/compare/maven-jar-plugin-2.6...maven-jar-plugin-3.1.1)

Signed-off-by: dependabot[bot] <support@dependabot.com>

He-Pin committed:
Merge pull request #15 from ReactivePlatform/dependabot/maven/org.apache.maven.plugins-maven-jar-plugin-3.1.1

Bump maven-jar-plugin from 2.6 to 3.1.1

He-Pin committed:
Merge pull request #17 from ReactivePlatform/dependabot/maven/org.apache.maven.plugins-maven-compiler-plugin-3.8.0

Bump maven-compiler-plugin from 3.3 to 3.8.0

He-Pin committed:
Merge pull request #16 from ReactivePlatform/dependabot/maven/org.codehaus.mojo-exec-maven-plugin-1.6.0

Bump exec-maven-plugin from 1.2.1 to 1.6.0

dependabot-support committed:
Bump netty-all from 4.1.32.Final to 4.1.33.Final

Bumps [netty-all](https://github.com/netty/netty) from 4.1.32.Final to 4.1.33.Final.
- [Release notes](https://github.com/netty/netty/releases)
- [Commits](https://github.com/netty/netty/compare/netty-4.1.32.Final...netty-4.1.33.Final)

Signed-off-by: dependabot[bot] <support@dependabot.com>

He-Pin committed:
Merge pull request #19 from ReactivePlatform/dependabot/maven/io.netty-netty-all-4.1.33.Final

Bump netty-all from 4.1.32.Final to 4.1.33.Final

dependabot-support committed:
Bump protobuf-java from 3.6.1 to 3.7.1

Bumps [protobuf-java](https://github.com/protocolbuffers/protobuf) from 3.6.1 to 3.7.1.
- [Release notes](https://github.com/protocolbuffers/protobuf/releases)
- [Changelog](https://github.com/protocolbuffers/protobuf/blob/master/generate_changelog.py)
- [Commits](https://github.com/protocolbuffers/protobuf/compare/v3.6.1...v3.7.1)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
Bump maven-surefire-plugin from 2.22.1 to 2.22.2

Bumps [maven-surefire-plugin](https://github.com/apache/maven-surefire) from 2.22.1 to 2.22.2.
- [Release notes](https://github.com/apache/maven-surefire/releases)
- [Commits](https://github.com/apache/maven-surefire/compare/surefire-2.22.1...surefire-2.22.2)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
Bump maven-failsafe-plugin from 2.22.1 to 2.22.2

Bumps [maven-failsafe-plugin](https://github.com/apache/maven-surefire) from 2.22.1 to 2.22.2.
- [Release notes](https://github.com/apache/maven-surefire/releases)
- [Commits](https://github.com/apache/maven-surefire/compare/surefire-2.22.1...surefire-2.22.2)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
Bump maven-compiler-plugin from 3.8.0 to 3.8.1

Bumps [maven-compiler-plugin](https://github.com/apache/maven-compiler-plugin) from 3.8.0 to 3.8.1.
- [Release notes](https://github.com/apache/maven-compiler-plugin/releases)
- [Commits](https://github.com/apache/maven-compiler-plugin/compare/maven-compiler-plugin-3.8.0...maven-compiler-plugin-3.8.1)

Signed-off-by: dependabot[bot] <support@dependabot.com>

He-Pin committed:
Merge pull request #23 from ReactivePlatform/dependabot/maven/com.google.protobuf-protobuf-java-3.7.1

Bump protobuf-java from 3.6.1 to 3.7.1

He-Pin committed:
Merge pull request #26 from ReactivePlatform/dependabot/maven/org.apache.maven.plugins-maven-surefire-plugin-2.22.2

Bump maven-surefire-plugin from 2.22.1 to 2.22.2

He-Pin committed:
Merge pull request #27 from ReactivePlatform/dependabot/maven/org.apache.maven.plugins-maven-failsafe-plugin-2.22.2

Bump maven-failsafe-plugin from 2.22.1 to 2.22.2

He-Pin committed:
Merge pull request #28 from ReactivePlatform/dependabot/maven/org.apache.maven.plugins-maven-compiler-plugin-3.8.1

Bump maven-compiler-plugin from 3.8.0 to 3.8.1

dependabot-support committed:
Bump maven-jar-plugin from 3.1.1 to 3.1.2

Bumps [maven-jar-plugin](https://github.com/apache/maven-jar-plugin) from 3.1.1 to 3.1.2.
- [Release notes](https://github.com/apache/maven-jar-plugin/releases)
- [Commits](https://github.com/apache/maven-jar-plugin/compare/maven-jar-plugin-3.1.1...maven-jar-plugin-3.1.2)

Signed-off-by: dependabot[bot] <support@dependabot.com>

He-Pin committed:
Merge pull request #29 from ReactivePlatform/dependabot/maven/org.apache.maven.plugins-maven-jar-plugin-3.1.2

Bump maven-jar-plugin from 3.1.1 to 3.1.2

dependabot-preview[bot] committed:
Bump protobuf-java from 3.7.1 to 3.8.0

Bumps [protobuf-java](https://github.com/protocolbuffers/protobuf) from 3.7.1 to 3.8.0.
- [Release notes](https://github.com/protocolbuffers/protobuf/releases)
- [Changelog](https://github.com/protocolbuffers/protobuf/blob/master/generate_changelog.py)
- [Commits](https://github.com/protocolbuffers/protobuf/compare/v3.7.1...v3.8.0)

He-Pin committed:
Merge pull request #31 from ReactivePlatform/dependabot/maven/com.google.protobuf-protobuf-java-3.8.0

Bump protobuf-java from 3.7.1 to 3.8.0

dependabot-preview[bot] committed:
Bump protobuf-java from 3.8.0 to 3.9.0

Bumps [protobuf-java](https://github.com/protocolbuffers/protobuf) from 3.8.0 to 3.9.0.
- [Release notes](https://github.com/protocolbuffers/protobuf/releases)
- [Changelog](https://github.com/protocolbuffers/protobuf/blob/master/generate_changelog.py)
- [Commits](https://github.com/protocolbuffers/protobuf/compare/v3.8.0...v3.9.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

He-Pin committed:
Merge pull request #35 from ReactivePlatform/dependabot/maven/com.google.protobuf-protobuf-java-3.9.0

Bump protobuf-java from 3.8.0 to 3.9.0

Commit:
upgrade to 4.1.42

LyWang1999 committed:
modify wrong code comments in chapter 5

Commit:
upgrade to 4.1.49

dependabot-preview[bot] committed:
Bump netty-all from 4.1.49.Final to 4.1.58.Final

Bumps [netty-all](https://github.com/netty/netty) from 4.1.49.Final to 4.1.58.Final.
- [Release notes](https://github.com/netty/netty/releases)
- [Commits](https://github.com/netty/netty/compare/netty-4.1.49.Final...netty-4.1.58.Final)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

He-Pin committed:
Merge pull request #78 from ReactivePlatform/dependabot/maven/io.netty-netty-all-4.1.58.Final

Bump netty-all from 4.1.49.Final to 4.1.58.Final

dependabot-preview[bot] committed:
Upgrade to GitHub-native Dependabot

dependabot-preview[bot] committed:
Bump protobuf-java from 3.9.0 to 3.17.3

Bumps [protobuf-java](https://github.com/protocolbuffers/protobuf) from 3.9.0 to 3.17.3.
- [Release notes](https://github.com/protocolbuffers/protobuf/releases)
- [Changelog](https://github.com/protocolbuffers/protobuf/blob/master/generate_changelog.py)
- [Commits](https://github.com/protocolbuffers/protobuf/compare/v3.9.0...v3.17.3)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

He-Pin committed:
!pom bump version to 4.1.75

mwwcdk committed:
第9章测试代码注释纠错

He-Pin committed:
Merge pull request #104 from mwwcdk/ChineseVersion

第9章测试代码注释纠错

He-Pin committed:
Merge pull request #100 from ReactivePlatform/dependabot/maven/com.google.protobuf-protobuf-java-3.17.3

Bump protobuf-java from 3.9.0 to 3.17.3

He-Pin committed:
Merge pull request #56 from LyWang1999/ChineseVersion

fix wrong code comments in chapter 5

He-Pin committed:
Merge pull request #92 from ReactivePlatform/dependabot/add-v2-config-file

Upgrade to GitHub-native Dependabot

He-Pin committed:
!pom bump java version to 1.8

He-Pin committed:
feat: upgrade Netty from 4.1.75.Final to 4.2.15.Final

Motivation:
Netty 4.2 带来了多项重要改进，包括弃用 NioEventLoopGroup 等
传输特定 EventLoopGroup，改用 MultiThreadIoEventLoopGroup + IoHandlerFactory
统一模型。需要同步升级示例代码以保持与实际 API 一致。

Modification:
- 升级 pom.xml 中 netty.version 至 4.2.15.Final
- 所有 NioEventLoopGroup 替换为 MultiThreadIoEventLoopGroup(NioIoHandler.newFactory())
- 保留 4.1 旧写法（注释形式），便于读者对照学习
- OioEventLoopGroup 保留使用并标注 @Deprecated（OIO 传输整体弃用）
- SelfSignedCertificate 标注弃用说明
- 翻译多处遗漏的英文注释（Listing → 代码清单，
  "get reference form somewhere" → "从某处获取引用" 等）

Result:
- 编译通过，所有示例代码与 Netty 4.2 API 对齐
- 中文翻译更完整，消除遗留英文注释
- 读者可同时看到 4.1 和 4.2 两种写法
